go/engine/login.go

```diff
@@ -140,7 +140,7 @@ func (e *Login) Run(m libkb.MetaContext) (err error) {
 	// clear out any existing session:
 	m.Debug("clearing any existing login session with Logout before loading user for login")
 	// If the doUserSwitch flag is specified, we don't want to kill the existing session
-	err = m.LogoutCurrentUserWithSecretKill(!e.doUserSwitch)
+	err = m.LogoutWithOptions(libkb.LogoutOptions{KeepSecrets: e.doUserSwitch})
```

Comment:
there is already `LogoutKeepSecrets` in `checkLoggedInAndNotRevoked` that's called. What flow requires logging out again here?

Reply:
It only does that if LoggedInWrongUser error right? not always?

Reply:
you're correct, there are two cases in which `false, nil` is returned from `checkLoggedInAndNotRevoked` in which we would get this far, and only one of them (`LoggedInWrongUserError` logs out, `NoActiveDeviceError` doesn't).